### PR TITLE
Add more tests

### PR DESCRIPTION
**Pull request by ipwnponies**
Adding more tests for the API.

The tests themselves are that interesting, it's the documentation of the API responses that will really help to speed up development.

The commits are per-endpoint test classes, should be easier to review in that scoped context.

### CODE REVIEW

**Review by edmundmok — APPROVED** [review]
Looks good at a glance, thanks for adding these tests!

**Comment by ipwnponies** [comment]
Added the last set of tests, this will document all the URL responses. We can now develop and iterate on features, without needing an account nor depending on the response (which changes depending on time of day, i.e. kitchen is closed all weekend).

**Review by edmundmok — APPROVED** [review]
New commits look good too, thanks!

File: mealpy/mealpy.py [hunk 1]
```diff
@@ -68,6 +68,9 @@ def login(self, user, password):
             'password': password,
         }
         request = self.session.post(LOGIN_URL, data=json.dumps(data))
```
Comment (dbgrigsby) [inline mealpy/mealpy.py]: This change breaks the program. Not having the json.dumps causes Mealpal to return a bad request message for all reservations.

**Pull request merged**